src/client/actions/GuildMemberUpdate.js

```diff
@@ -27,7 +27,7 @@ class GuildMemberUpdateAction extends Action {
          * @param {GuildMember} oldMember The member before the update
          * @param {GuildMember} newMember The member after the update
          */
-        if (shard.status === Status.READY) client.emit(Events.GUILD_MEMBER_UPDATE, old, member);
+        if (!member.equals(old) && shard.status === Status.READY) client.emit(Events.GUILD_MEMBER_UPDATE, old, member);
```

Comment:
Swap the order of the checks, the last one is a lot faster and can discard a much larger chain of operations.